cpp/src/parquet/bloom_filter_writer.cc

```
@@ -0,0 +1,305 @@
+    } else {
+      bloom_filter_->Hashes(values, batch_size, hashes.data());
+    }
+    bloom_filter_->InsertHashes(hashes.data(), batch_size);
+  }
+}
+
+template <>
+void BloomFilterWriter<BooleanType>::Update(const bool*, int64_t) {
+  if (!bloom_filter_enabled()) {
+    return;
+  }
+  throw ParquetException("Bloom filter is not supported for boolean type");
+}
+
+template <typename ParquetType>
+void BloomFilterWriter<ParquetType>::UpdateSpaced(const T* values, int64_t num_values,
+                                                  const uint8_t* valid_bits,
+                                                  int64_t valid_bits_offset) {
+  if (!bloom_filter_enabled()) {
+    return;
+  }
+
+  std::array<uint64_t, kHashBatchSize> hashes;
+  ::arrow::internal::VisitSetBitRunsVoid(
+      valid_bits, valid_bits_offset, num_values, [&](int64_t position, int64_t length) {
+        for (int64_t i = 0; i < length; i += kHashBatchSize) {
+          auto batch_size = static_cast<int>(std::min(kHashBatchSize, length - i));
+          if constexpr (std::is_same_v<ParquetType, FLBAType>) {
+            bloom_filter_->Hashes(values + i + position, descr_->type_length(),
+                                  batch_size, hashes.data());
+          } else {
+            bloom_filter_->Hashes(values + i + position, batch_size, hashes.data());
+          }
+          bloom_filter_->InsertHashes(hashes.data(), batch_size);
+        }
+      });
+}
+
+template <>
+void BloomFilterWriter<BooleanType>::UpdateSpaced(const bool*, int64_t, const uint8_t*,
+                                                  int64_t) {
+  if (!bloom_filter_enabled()) {
+    return;
+  }
+  throw ParquetException("Bloom filter is not supported for boolean type");
+}
+
+template <typename ParquetType>
+void BloomFilterWriter<ParquetType>::Update(const ::arrow::Array& values) {
+  ::arrow::Unreachable("Update for non-ByteArray type should be unreachable");
```

Why? Can you please at least raise a proper exception?